bumpversion/version_part.py

```diff
@@ -63,8 +63,37 @@ def __repr__(self):
             self.value
         )
 
+    def __hash__(self):
+        return hash(self.value)
+
+    def _compare(self, other, method):
+        if self.config.function_cls is not other.config.function_cls:
+            raise TypeError("Versions use different part specific configuration, cant compare them.")
+        if self.config.function_cls is NumericFunction:
+            return method(self.value, other.value)
+        else:
+            # Compare order
+            idx1 = self.config.function.index(self.value)
+            idx2 = other.config.function.index(other.value)
+            return method(idx1, idx2)
+
     def __eq__(self, other):
-        return self.value == other.value
+        return self._compare(other, lambda s, o: s == o)
+
+    def __lt__(self, other):
+        return self._compare(other, lambda s, o: s < o)
+
+    def __le__(self, other):
+        return self._compare(other, lambda s, o: s <= o)
+
+    def __ge__(self, other):
+        return self._compare(other, lambda s, o: s >= o)
+
+    def __gt__(self, other):
+        return self._compare(other, lambda s, o: s > o)
+
+    def __ne__(self, other):
+        return self._compare(other, lambda s, o: s != o)
```

Comment:
Can this be simplified to `return not self == other`? See https://stackoverflow.com/questions/4352244/python-should-i-implement-ne-operator-based-on-eq#30676267 as well.

Reply:
OK